pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -1204,7 +1205,47 @@ func (i *InstallConfig) Image() string {
 }
 
 // Disk implements the config.Provider interface.
+//nolint:gocyclo
 func (i *InstallConfig) Disk() string {
+	if i.InstallDiskSelector != nil {
+		selector := i.InstallDiskSelector
+
+		matchers := []disk.Matcher{}
+		if selector.Size != nil {
+			matchers = append(matchers, selector.Size.matcher)
+		}
+
+		if selector.UUID != "" {
+			matchers = append(matchers, disk.WithUUID(selector.UUID))
+		}
+
+		if selector.WWID != "" {
+			matchers = append(matchers, disk.WithWWID(selector.WWID))
+		}
+
+		if selector.Model != "" {
+			matchers = append(matchers, disk.WithModel(selector.Model))
+		}
+
+		if selector.Name != "" {
+			matchers = append(matchers, disk.WithName(selector.Name))
+		}
+
+		if selector.Serial != "" {
+			matchers = append(matchers, disk.WithSerial(selector.Serial))
+		}
+
+		if selector.Modalias != "" {
+			matchers = append(matchers, disk.WithModalias(selector.Modalias))
+		}
+
+		disk, err := disk.Find(matchers...)
```

Comment:
I wonder if this code should be outside of the config provider really?

Like we have method like `DiskMatchers() []diskMatcher` which returns matchers, if that list is not empty, we prefer it over `InstallDisk`, but the actual `disk.Find()` call is somewhere in the sequencer (tasks?), so we could print error message saying if matching failed, etc.?

Reply:
Yep, let's change that.

Reply:
Ah wait, I remember why I moved that here.
We also have validation module, which get the disk using `Disk` method.

Still can split it though. Should be a common method for `v1alpha1_validation.go` and the sequener.